walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,61 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.function.Predicate;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<Event> eventsList = new LinkedList<>();
+    List<Event> eventsListOptional = new LinkedList<>();
+    for (Event event : events) {
+      // add event to list if event and request have attendees in common
+      if (!Collections.disjoint(event.getAttendees(), request.getAttendees())) {
+        eventsList.add(event);
+        eventsListOptional.add(event);
```

Comment:
Why does the event get added to the optional list here?

Reply:
my thought process was that the mandatory events list must have all mandatory attendees while the optional attendees events list must have all optional attendees and might have mandatory attendees but could still be optional as a whole.

Reply:
Hm, I'm still confused, but I've already forgotten what exactly this method is meant to do. Can you add a java doc to explain how your algorithm works?

Reply:
I added the javadoc and more comments